src/sentry/tasks/find_missing_codemappings.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from ast import Tuple
+from datetime import timedelta
+from typing import List
+
+from django.utils import timezone
+
+from sentry.db.models.fields.node import NodeData
+from sentry.models import Project
+from sentry.models.group import Group
+from sentry.models.organization import Organization, OrganizationStatus
+from sentry.tasks.base import instrumented_task
+from sentry.utils.safe import get_path
+
+PREFERRED_GROUP_OWNERS = 1
+PREFERRED_GROUP_OWNER_AGE = timedelta(days=7)
+
+logger = logging.getLogger("tasks.commit_context")
+
+
+@instrumented_task(
+    name="sentry.tasks.find_missing_codemappings",
+    queue="find_missing_codemappings",
+    max_retries=0,  # if we don't backfill it this time, we'll get it the next time
+)
+def find_missing_codemappings(**kwargs):
+    organizations = kwargs.get(
+        "organizations", Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+    )
+
+    filename_maps = {}
+    for org in organizations:
+        projects = Project.objects.filter(organization=org, first_event__isnull=False)
+
+        projects = [
+            project
+            for project in projects
+            if Group.objects.filter(
+                project=project, last_seen__gte=timezone.now() - timedelta(days=7)
+            ).exists()
+        ]
+
+        project_file_map = {project.slug: get_all_filenames(project) for project in projects}
+        filename_maps[org.slug] = project_file_map
+    return filename_maps
+
+
+def get_all_filenames(project):
+    groups = Group.objects.filter(
+        project=project, last_seen__gte=timezone.now() - timedelta(days=14)
+    )
+
+    filenames = set()
+    for group in groups:
+        event = group.get_latest_event()
+        is_python_project, fn = get_filenames(project, event.data)
+        # Note: this will skip any possible python files in a project if we find a single
+        # #  non-".py" file in the stacktrace. Is this too unforgiving?
+        if not is_python_project:
```

Comment:
Is it possible for the stacktrace to have non-python files? If so, this will skip the python files we found for the project as well.